app/controllers/staff/speakers_controller.rb

```diff
@@ -1,60 +1,51 @@
 class Staff::SpeakersController < Staff::ApplicationController
   decorates_assigned :speaker
-  before_filter :set_proposal, only: [:new, :create, :destroy]
+  before_action :set_program_session, only: [:new, :create]
 
   def index
-    render locals: {
-             proposals: @event.proposals.includes(speakers: :user).decorate
-           }
+    @program_speakers = current_event.speakers.in_program
   end
 
   def new
     @speaker = Speaker.new
-    @user = @speaker.build_user
-    @proposal = Proposal.find_by(uuid: params[:proposal_uuid])
   end
 
   def create
     s_params = speaker_params
-    user = User.find_by(email: s_params.delete(:speaker_email))
-    if user
-      if user.speakers.create(s_params.merge(proposal: @proposal))
-        flash[:success] = "Speaker was added to this proposal"
-      else
-        flash[:danger] = "There was a problem saving this speaker"
-      end
+    @speaker = @program_session.speakers.create(s_params.merge(event: current_event))
+    if @speaker.save
+      flash[:success] = "#{@speaker.name} has been added to #{@program_session.title}"
     else
-      flash[:danger] = "Could not find a user with this email address"
+      flash[:danger] = "There was a problem saving this speaker."
     end
-    redirect_to event_staff_proposal_path(event, @proposal)
+    redirect_to event_staff_speakers_path(current_event)
```

Comment:
Just noticed this while testing. If there's a problem on the form, the user is redirected back to the speakers index instead of the form. The update action below does it right vv